### ISSUE TITLE
deadbeef-git: update to latest

### ISSUE THREAD

**Issue by huglovefan** [huglovefan/srcpkgs#1]
i was having build failures related to the `strdupa` call [here](https://github.com/DeaDBeeF-Player/deadbeef/blob/2c52eea807788908cc7341feedbcdf617773090c/plugins/adplug/adplug-db.cpp#L99) (something about pointers) which i solved by using the version before that commit

apparently `strdupa` is an evil gnu extension, deadbeef does define a fallback [here](https://github.com/DeaDBeeF-Player/deadbeef/blob/26dba6d4d928b0872c3fcc4a5cbf2d91c61e3187/strdupa.h) but it did not seem to work

**Comment by huglovefan** [comment]
okay

there are other occurrences of the `const char *uri = strdupa (deadbeef->pl_find_meta (it, ":URI"));` pattern in deadbeef but this is the only one in c++ which has some stricter rules regarding pointers

the deadbeef fallback is fine and works but it's not used because `stdrupa` is already defined [here](https://git.musl-libc.org/cgit/musl/tree/include/string.h?id=21a172dd36cae7a08492fd3a7500d7bf0daee13e#n88) in `string.h`

the definition looks like this:

```c
#define strdupa(x) strcpy(alloca(strlen(x)+1),x)
```

it doesn't cast the result and neither does deadbeef so you get an error like this in c++ mode:

```
test.cpp:7:21: error: invalid conversion from 'void*' to 'char*' [-fpermissive]
  const char *test = strdupa("hi");
                     ^
```

**Comment by huglovefan** [comment]
fixed in 73b3f4b8fa9f1ff8dd195fe18553f74f3e1ad486 (uses deadbeef's `strdupa` on musl)
